source/server/admin/stats_params.h

```diff
@@ -32,6 +35,7 @@ struct StatsParams {
   StatsFormat format_{StatsFormat::Text};
   std::string filter_string_;
   absl::optional<std::regex> filter_;
+  std::shared_ptr<re2::RE2> safe_filter_;
```

Comment:
Why shared_ptr and not unique_ptr? Or optional<re2::RE2> like the `filter_`?

Reply:
will add a comment: in the current usage the StatsParams object is copied into a Request and I think also a Render object, and unique_ptr makes it non-copyable.

Reply:
SG. It would only get copied once per request but still it might as well parallel the RE2.

Reply:
done